Mathlib/Analysis/Distribution/SchwartzSpace.lean

```diff
@@ -1215,4 +1247,112 @@ def toZeroAtInftyCLM : 𝓢(E, F) →L[𝕜] C₀(E, F) :=
 
 end ZeroAtInfty
 
+section Lp
+
+/-! ### Inclusion into L^p space -/
+
+open MeasureTheory
+open scoped NNReal ENNReal
+
+variable [NormedAddCommGroup D] [MeasurableSpace D] [MeasurableSpace E] [OpensMeasurableSpace E]
+  [NormedField 𝕜] [NormedSpace 𝕜 F] [SMulCommClass ℝ 𝕜 F]
+
+variable (𝕜 F) in
+/-- The `L^p` norm of a Schwartz function is controlled by a finite family of Schwartz seminorms.
+
+The maximum index `k` and the constant `C` depend on `p` and `μ`.
+-/
+theorem eLpNorm_le_seminorm (p : ℝ≥0∞) (μ : Measure E := by volume_tac)
+    [hμ : μ.HasTemperateGrowth] :
+    ∃ (k : ℕ) (C : ℝ≥0), ∀ (f : 𝓢(E, F)), eLpNorm f p μ ≤
+      C * ENNReal.ofReal ((Finset.Iic (k, 0)).sup (schwartzSeminormFamily 𝕜 E F) f) := by
+  -- Apply Hölder's inequality `‖f‖_p ≤ ‖f₁‖_p * ‖f₂‖_∞` to obtain the `L^p` norm of `f = f₁ • f₂`
+  -- using `f₁ = (1 + ‖x‖) ^ (-k)` and `f₂ = (1 + ‖x‖) ^ k • f x`.
+  rcases hμ.exists_eLpNorm_lt_top p with ⟨k, hk⟩
+  refine ⟨k, (eLpNorm (fun x ↦ (1 + ‖x‖) ^ (-k : ℝ)) p μ).toNNReal * 2 ^ k, fun f ↦ ?_⟩
+  have h_one_add (x : E) : 0 < 1 + ‖x‖ := lt_add_of_pos_of_le zero_lt_one (norm_nonneg x)
+  calc eLpNorm (⇑f) p μ
+  _ = eLpNorm ((fun x : E ↦ (1 + ‖x‖) ^ (-k : ℝ)) • fun x ↦ (1 + ‖x‖) ^ k • f x) p μ := by
+    refine congrArg (eLpNorm · p μ) (funext fun x ↦ ?_)
+    simp [Real.rpow_neg (h_one_add x).le, (h_one_add x).ne']
+  _ ≤ eLpNorm (fun x ↦ (1 + ‖x‖) ^ (-k : ℝ)) p μ * eLpNorm (fun x ↦ (1 + ‖x‖) ^ k • f x) ⊤ μ := by
+    refine eLpNorm_smul_le_eLpNorm_mul_eLpNorm_top p _ ?_
+    refine Continuous.aestronglyMeasurable ?_
+    exact .rpow_const (.add continuous_const continuous_norm) fun x ↦ .inl (h_one_add x).ne'
+  _ ≤ eLpNorm (fun x ↦ (1 + ‖x‖) ^ (-k : ℝ)) p μ *
+      (2 ^ k * ENNReal.ofReal (((Finset.Iic (k, 0)).sup (schwartzSeminormFamily 𝕜 E F)) f)) := by
+    gcongr _ * ?_
```

Comment:
```suggestion
    gcongr
```
`gcongr` is clever enough to match things by itself most of the time. The advanced syntax is only useful in more specific situations where it gets it wrong.